inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -169,13 +169,20 @@ ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(co
     std::unordered_map<std::string, InferenceEngine::Parameter> multiNetworkConfig;
     multiNetworkConfig.insert(*priorities);
 
+    std::mutex mutex;
     DeviceMap<ExecutableNetwork> executableNetworkPerDevice;
+    std::vector<std::future<void>> loads;
     for (auto& p : metaDevices) {
-        auto & deviceName = p.deviceName;
-        auto & deviceConfig = p.config;
-        executableNetworkPerDevice.insert({ deviceName, GetCore()->LoadNetwork(network, deviceName, deviceConfig) });
-        multiNetworkConfig.insert(deviceConfig.begin(), deviceConfig.end());
-    }
+        loads.push_back(std::async(std::launch::async, [&]() {
```

Comment:
It is better to use IE Executor + Executors cache? Even without TBB it preserve the count of new threads.
```c++
ExecutorManager::getInstance()->getIdleCPUStreamsExecutor(streamsExecutorConfig);
```

Reply:
`ITaskExecutor::runAndWait` do what you need